Zend/zend_execute.c

```diff
@@ -1352,11 +1354,27 @@ static void zend_assign_to_string_offset(zval *str, zend_long offset, zval *valu
 	if (Z_TYPE_P(value) != IS_STRING) {
 		zend_string *tmp = zval_get_string(value);
 
-		Z_STRVAL_P(str)[offset] = ZSTR_VAL(tmp)[0];
+		string_len = ZSTR_LEN(tmp);
+		ch = ZSTR_VAL(tmp)[0];
 		zend_string_release(tmp);
 	} else {
-		Z_STRVAL_P(str)[offset] = Z_STRVAL_P(value)[0];
+		string_len = Z_STRLEN_P(value);
+		ch = Z_STRVAL_P(value)[0];
 	}
+
+	/* If input string is empty, change nothing and raise a warning */
+
+	if (string_len == 0) {
+		zend_error(E_WARNING, "Cannot assign an empty string to a string offset");
+		zend_string_release(old_str);
+		if (result) {
+			ZVAL_NULL(result);
```

Comment:
Unsure about this return value, we should cross check it with other assignment failure cases. I think we should be either returning an empty string here (the RHS) or the character at the modified offset of the string (the "new" LHS).